tests/unit/client.py

```diff
@@ -13,140 +13,139 @@
+from hil.client.client import Client, HTTPClient, HTTPResponse
+from hil.test_common import config_testsuite, config_merge, \
+    fresh_database, fail_on_log_warnings
+from hil.model import db
+from hil import config, deferred, server
 
 import json
-import os
 import pytest
-import requests
-import sys
-import tempfile
-import time
 
-from subprocess import check_call, Popen
+from urlparse import urlparse
+from base64 import urlsafe_b64encode
+from passlib.hash import sha512_crypt
 
-ep = "http://127.0.0.1:8000" or os.environ.get('HIL_ENDPOINT')
-username = "hil_user" or os.environ.get('HIL_USERNAME')
-password = "hil_pass1234" or os.environ.get('HIL_PASSWORD')
+ep = "http://127.0.0.1:8000"
+username = "hil_user"
+password = "hil_pass1234"
 
-http_client = RequestsHTTPClient()
-http_client.auth = (username, password)
+
+class FlaskHTTPClient(HTTPClient):
+
+    def __init__(self):
+        self._flask_client = app.test_client()
+
+    def request(self, method, url, data=None, params=None):
+
+        # Flask doesn't provide a straightforward way to do basic auth,
+        # but it's not actually that complicated:
+        auth_header = 'Basic ' + urlsafe_b64encode(username + ':' + password)
+
+        resp = self._flask_client.open(
+            method=method,
+            headers={'Authorization': auth_header},
+            # flask expects just a path, and assumes
+            # the host & scheme:
+            path=urlparse(url).path,
+            data=data,
+            query_string=params,
+        )
+        return HTTPResponse(status_code=resp.status_code,
+                            headers=resp.headers,
+                            content=resp.get_data())
+
+http_client = FlaskHTTPClient()
 C = Client(ep, http_client)  # Initializing client library
 MOCK_SWITCH_TYPE = 'http://schema.massopencloud.org/haas/v0/switches/mock'
 OBM_TYPE_MOCK = 'http://schema.massopencloud.org/haas/v0/obm/mock'
 OBM_TYPE_IPMI = 'http://schema.massopencloud.org/haas/v0/obm/ipmi'
 
 
-class Test_ClientBase:
-    """Tests client initialization and object_url creation. """
-
-    def test_init_error(self):
-        with pytest.raises(TypeError):
-            x = ClientBase()
-
-    def test_object_url(self):
-        x = ClientBase(ep, 'some_base64_string')
-        y = x.object_url('abc', '123', 'xy23z')
-        assert y == 'http://127.0.0.1:8000/abc/123/xy23z'
-
-# For testing the client library we need a running HIL server, with dummy
-# objects populated. Following classes accomplish that end.
-# It shall:
-#       1. Configures hil.cfg
-#       2. Instantiates a database
-#       3. Starts a server on an arbitary port
-#       4. Populates hil with dummy objects
-#       5. tears down the setup in a clean fashion.
+fail_on_log_warnings = pytest.fixture(fail_on_log_warnings)
+fresh_database = pytest.fixture(fresh_database)
 
-db_dir = None
 
+@pytest.fixture
+def dummy_verify():
+    """replace sha512_crypt.verify with something faster (albeit broken).
```

Comment:
I don't really understand how this works. 

1. What was calling it (db-auth?) 
2. How are we over-riding/mocking the original ?
3. Why do we keep `sha512_crypt.verify` in old and then restore it after yield? It still works if I don't do this.